### PR TITLE
Refactor check_* steps into filter modules

### PR DESCRIPTION
**Pull request by foxcpp**
Keeping low-level pipeline dispatching code only for low-level dispatching.

Will cause merge conflicts with #61.

### CODE REVIEW

**Comment by emersion** [comment]
Needs rebase.

**Comment by foxcpp** [comment]
Rebased.

File: function_filter.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"github.com/emersion/maddy/module"
+)
+
+type FuncFilter func(ctx *module.DeliveryContext, msg io.Reader) (io.Reader, error)
+
+// FunctionFilter is a wrapper to help create simple filter functions that
+// don't need any state or configuration.
+//
+// See RegisterFunctionFilter.
+type FunctionFilter struct {
```
Comment (emersion) [inline function_filter.go]: This doesn't need to be exported

**Comment by emersion** [comment]
Thanks!

**Pull request merged**